pycbc/detector.py

```diff
@@ -444,3 +445,138 @@ def time_delay_from_earth_center(self, right_ascension, declination, t_gps):
             np.array([np.float32(earth.x), np.float32(earth.y),
                       np.float32(earth.z)]), right_ascension,
             declination, t_gps)
+
+    def interpolate_hij(self, hp, hc):
+        """Returns the values of hp, hc after
+            interpoalting it to the desired time
+        Parameters
+        ----------
+        ref_time : numpy.ScalarType
+            GPS time
+        hp, hc : pycbc.types.TimeSeries
+            Plus and cross polarizations in order
+        time : numpy.ndarray (3,1)
+            GPS time
+        Returns
+        -------
+        numpy.ndarray shape (3,1)
+            The values of hij at the required time
+        """
+        fhp = interpolate.interpolate.interp1d(hp.sample_times, hp)
```

Comment:
@SSingh087 You seem to be hard coding the waveform to be sampled at a fixed range, just one that's higher than the input. Why don't you just return the interpolating functions and delete lines 467-470?